### PR TITLE
RequirementMachine: Fix terrible bug in minimization of protocol inheritance

### PR DESCRIPTION
**Pull request by slavapestov**
Fixes rdar://problem/90402503.

### CODE REVIEW

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please test source compatibility

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please test source compatibility

File: lib/AST/RequirementMachine/MinimalConformances.cpp [hunk 1]
```diff
@@ -177,15 +179,17 @@ class MinimalConformances {
...
-  // Maps a conformance rule to a list of paths. Each path in the list is a unique
-  // derivation of the conformance in terms of other conformance rules.
+  // Maps a conformance rule in the current minimization domain to a list of paths.
+  // Each path in the list is a unique derivation of the conformance in terms of
+  // other conformance rules.
   llvm::MapVector<unsigned, std::vector<SmallVector<unsigned, 2>>> ConformancePaths;
 
-  // The set of conformance rules which are protocol refinements, that is rules of
-  // the form [P].[Q] => [P].
+  // The set of conformance rules (from all minimization dmains) which are protocol
```
Comment (ibrahimkteish) [inline lib/AST/RequirementMachine/MinimalConformances.cpp:187]:
```suggestion
  // The set of conformance rules (from all minimization domains) which are protocol
```

**Comment by slavapestov** [comment]
@swift-ci Please smoke test Linux

**Comment by slavapestov** [comment]
@swift-ci Please smoke test Linux

**Pull request merged**